src/Symfony/Component/HtmlSanitizer/Visitor/Node/Node.php

```diff
@@ -57,7 +60,11 @@ public function addChild(NodeInterface $node): void
     public function render(): string
     {
         if (!$this->children) {
-            return '<'.$this->tagName.$this->renderAttributes().' />';
+            if (\in_array($this->$tagName, self::VOID_ELEMENTS)) {
+                return '<'.$this->tagName.$this->renderAttributes().' />';
+            }
+
+            return '<'.$this->tagName.$this->renderAttributes().'></'.$this->tagName.'>';
```

Comment:
we could drop this line and merge both ifs above as:
`if (!$this->children && \in_array($this->$tagName, self::VOID_ELEMENTS, true)) {`

Reply:
added some comments, the returns are different for each if. unless i misunderstand what you mean